Dashboards: Remove panel plugin provider from migrations

ryantxu opened the pull request:
In https://github.com/grafana/grafana/pull/108416, we added a `PanelPluginInfoProvider` to the migrator -- we should avoid this unless it is absolutely necessary and is actually used to migrate the panels.

From the current usage it is not necessary and would put real values into an invalid state

ryantxu committed:
remove panel provider

ryantxu committed:
use 1.0.0

ryantxu committed:
cleanup

ryantxu committed:
singlestat

ryantxu commented on apps/dashboard/pkg/migration/schemaversion/v28.go:
```diff
@@ -145,13 +134,7 @@ func (m *v28Migrator) migrateSinglestatPanel(panel map[string]interface{}) error
-	}
-
-	panel["pluginVersion"] = m.statPanelVersion
+	panel["pluginVersion"] = "1.0.0" // FIXME... static version at this time
```
^^^ FIXME

konsalex commented on apps/dashboard/pkg/migration/schemaversion/migrations.go:
```diff
@@ -48,11 +41,11 @@ func GetMigrations(dsInfoProvider DataSourceInfoProvider, panelProvider PanelPlu
 		25: V25,
 		26: V26,
 		27: V27,
-		28: V28(panelProvider),
```
Technically we could even refactor v28 migrator to not have be a factory, and follow all the other migrators pattern except the ones with datasources

dprokop commented on apps/dashboard/pkg/migration/schemaversion/v24.go:
```diff
@@ -224,12 +210,7 @@ func (m *v24Migrator) migrate(dashboard map[string]interface{}) error {
-			return NewMigrationError("table panel plugin not found when migrating dashboard to schema version 24", 24, LATEST_VERSION, "V24")
-		}
-		panelMap["pluginVersion"] = tablePanelPlugin.Version
+		panelMap["pluginVersion"] = "1.0.0" // FIXME: what was the grafanaVersion at schema v24?
```
it's 7.0.0 for v24 and 8.0.0 for v28

dprokop committed:
Merge main into dashboard-migraitons-remove-panel-provider

dprokop committed:
Use dynamic Grafana version for panel plugin migrations

- Replace hardcoded plugin versions in V24 and V28 migrations with setting.BuildVersion
- Add fallback to current dev version (12.2.0-pre) for test environments
- Update test expectations and mock plugin versions to match dynamic versioning
- Regenerate migration test output files with current versions

dprokop committed:
Refactor V28 migration from factory function to direct function

ryantxu commented on apps/dashboard/pkg/migration/schemaversion/v24.go:
```diff
@@ -225,12 +213,14 @@ func (m *v24Migrator) migrate(_ context.Context, dashboard map[string]interface{
-		if tablePanelPlugin.ID == "" {
-			return NewMigrationError("table panel plugin not found when migrating dashboard to schema version 24", 24, LATEST_VERSION, "V24")
+		// Use dynamic Grafana version for pluginVersion
+		// In production: setting.BuildVersion contains actual build version
+		// In tests: setting.BuildVersion is empty, so we use fallback
+		pluginVersion := setting.BuildVersion
+		if pluginVersion == "" {
+			pluginVersion = "12.2.0-pre" // fallback for development/test environments
 		}
-		panelMap["pluginVersion"] = tablePanelPlugin.Version
+		panelMap["pluginVersion"] = pluginVersion
```
```suggestion
		// The grafana version that matches the hardcoded autoMigrate plugins
		panelMap["pluginVersion"] = "12.1.0"
```

ryantxu committed:
Merge remote-tracking branch 'origin/main' into dashboard-migraitons-remove-panel-provider

ryantxu committed:
update snapshot version

ryantxu committed:
remove stray file

ryantxu committed:
update snapshot version

ryantxu committed:
add the constant to frontend test also

ryantxu committed:
ignore static

ryantxu committed:
import order

dprokop reviewed (approved):
Thanks Ryan!
